packages/babel-plugin-transform-decorators-2/src/index.js

```diff
@@ -0,0 +1,183 @@
+import syntaxDecorators2 from "babel-plugin-syntax-decorators-2";
+
+//TODO: will have to check for dot (.) access to reserved keywords in decorator members
+//TODO: check if we need the 'define' + 'Property' and 'ke' + 'ys' hacks as seen in
+//https://github.com/loganfsmyth/babel-plugin-transform-decorators-legacy/blob/master/src/index.js#L69
+
+/** manual testing code
+**/
+export default function({ types: t }) {
+  // converts [(expression)] to [(let key = (expression))] if expression is impure
+  // so as to avoid recomputation when the key is needed later
+  function addKeysToComputedMembers(path) {
+    const body = path.get("body.body");
+
+    for (const member of body) {
+      if (member.node.computed && member.get("key").isPure()) {
+        const parent = path.findParent(p => p.parentPath.isBlock());
+        const ref = member.scope.generateUidIdentifier("key");
```

Comment:
IIRC there is a method which generates an uid using a node (e.g. `computed.property + 1` -> `computed$propertyPlus1`), so we have more descriptive names and avoid `_key26739`